apple/validator.go

```diff
@@ -65,6 +65,24 @@ func NewWithURL(validationURL string, revokeURL string) *Client {
 	return client
 }
 
+// NewWithClient creates a Client object with a custom http client
+func NewWithClient(httpClient *http.Client) *Client {
+	if httpClient == nil {
+		httpClient = http.DefaultClient
+	}
+
+	if httpClient.Timeout < 5*time.Second {
```

Comment:
Im curious to understand why we are enforcing a min of 5s timeout. If the user wants to pass in a custom client, shouldn't they be able to use whatever timeout they want to (even if its less than 5 seconds)?